ledger/ledger-on-sql/src/app/scala/com/daml/ledger/on/sql/SqlLedgerFactory.scala

```diff
@@ -18,16 +18,16 @@ import com.daml.ledger.participant.state.kvutils.caching._
 import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
 import com.daml.lf.engine.Engine
 import com.daml.logging.LoggingContext
-import com.daml.platform.configuration.LedgerConfiguration
+import com.daml.platform.configuration.InitialLedgerConfiguration
 import scopt.OptionParser
 
 object SqlLedgerFactory extends LedgerFactory[ReadWriteService, ExtraConfig] {
   override val defaultExtraConfig: ExtraConfig = ExtraConfig(
     jdbcUrl = None
   )
 
-  override def ledgerConfig(config: Config[ExtraConfig]): LedgerConfiguration =
-    super.ledgerConfig(config).copy(initialConfigurationSubmitDelay = Duration.ZERO)
+  override def initialLedgerConfig(config: Config[ExtraConfig]): InitialLedgerConfiguration =
+    super.initialLedgerConfig(config).copy(delayBeforeSubmitting = Duration.ofMillis(500))
```

Comment:
Same here. Maybe we can extract to a constant and reuse if it has the same meaning

Reply:
We had these constants defined in a common place ([here](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/configuration/LedgerConfiguration.scala#L46)). Thinking about it again, there should be 3 different common scenarios:

1. Ledger without persistence. The submission delay should be zero, as each application start will create a new ledger.
2. Participant where the index database is always up to date. The submission delay should be very short, as we only need to wait one roundtrip to the database to see whether there is a configuration.
3. Participant where the index database may have to be recreated from a distributed ledger. The submission delay should be a bit longer, to allow the indexer to ingest a few updates from the ReadService.

If the delay is too short, each participant start may produce a "Configuration change rejected" update on the ledger, which should not affect anything. Maybe we could review those constants and add them to `object InitialLedgerConfiguration`?

Reply:
I think the constants we had in `LedgerConfiguration` were quite misleading; ledgers can vary quite a bit, and we can't come up with decent values for all of them. I have extracted a couple of useful constants specifically for KV and added them to the KV `Config` object, but I don't think these should necessarily extend outwards to other driver implementations.

For example, I would expect an implementation based on Ethereum to be working in minutes, not seconds, and it makes very little sense for a Canton-based driver to allow the participant to set a configuration at all. (In the near future, this behavior will be completely optional, and Canton will not provide an `InitialLedgerConfiguration`.)